File: 1448. Count Good Nodes in Binary Tree/Solution1448.java
class Solution1448 {
    private int numGoodNodes = 0;
    
    public int goodNodes(TreeNode root) {
        dfs(root, Integer.MIN_VALUE);
        return numGoodNodes;
    }
    
    private void dfs(TreeNode node, int maxSoFar) {
        if (maxSoFar <= node.val) {
            numGoodNodes++;
        }
        if (node.left != null) {
            dfs(node.left, Math.max(node.val, maxSoFar));
        }
        if (node.right != null) {
            dfs(node.right, Math.max(node.val, maxSoFar));
        }
    }
}

File: 1448. Count Good Nodes in Binary Tree/Solution1448.cpp
#include<vector>
#include<iostream>
using namespace std;

class Solution1448 {
public:
    int goodNodes(TreeNode* root) {
        vector<int> count(1,0);
        helper(root, INT_MIN, count);
        return count[0];
    }
    void helper(TreeNode* root, int maxSoFar, vector<int>& count) {
        if (root->val >= maxSoFar) {
            count[0]++;
            maxSoFar = root->val;
        }
        if (root->left) {
            helper(root->left, maxSoFar, count);
        }
        if (root->right) {
            helper(root->right, maxSoFar, count);
        }        
    }
};